packages/shared/src/hooks/utils/useGetShortUrl.ts

```diff
@@ -0,0 +1,45 @@
+import { useQueryClient } from '@tanstack/react-query';
+import request from 'graphql-request';
+import { useCallback } from 'react';
+import { graphqlUrl } from '../../lib/config';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { GET_SHORT_URL_QUERY } from '../../graphql/urlShortener';
+import { addTrackingQueryParams } from '../../lib/share';
+
+interface UseGetShortUrlResult {
+  getShortUrl: (url: string, cid?: string) => Promise<string>;
```

Comment:
I would make the props object based here.
Just easier for other devs to know what you pass/add/change order etc.

Reply:
I think if it's 2 parameters that we pass, one being optional, is it really that hard to figure out? Especially if you look at it in an IDE, you pretty much see immediately which parameter means what.

Reply:
Not with 2, but next week Lee will have to add a third one, then a fourth you know how these things go.
Especially if all become optional it's a pain to pass only the ones you want.

Reply:
Well, let's cross that bridge once we get there :) 

I think we can stick to the rule of - if it's 3 or more parameters, convert to object params, otherwise, feel free to use normal function params. 

Most of the projects I've worked in the past had similar logic applied to the linter, maybe we can do the same?

Reply:
I prefer to just always go object and add linting for that.
Don't see any negatives on that approach 🙏

It's non blocking for now, but will raise it for web guild.